internal/controllers/utils/utils_test.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2024 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package utils
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	oranv1alpha1 "github.com/openshift-kni/oran-o2ims/api/v1alpha1"
+)
+
+var suitescheme = scheme.Scheme
+
+func getFakeClientFromObjects(objs ...client.Object) (client.WithWatch, error) {
+	return fake.NewClientBuilder().WithScheme(suitescheme).WithObjects(objs...).WithStatusSubresource(&oranv1alpha1.ORANO2IMS{}).Build(), nil
+}
+
+func TestORANO2IMSControllerUtils(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Controller Utils Suite")
+}
+
+var _ = Describe("ExtensionUtils", func() {
+	It("Returns the string for all the extensions", func() {
+		extensions := []oranv1alpha1.Extension{
+			{
+				Key:   "key1",
+				Value: "value1",
+			},
+			{
+				Key:   "key2",
+				Value: "value2",
+			},
+		}
+
+		extString := ExtensionToString(extensions)
+		Expect(extString).To(Equal("key1: value1, key2: value2"))
+	})
+
+	It("Returns empty string when there are no extensions", func() {
+		extensions := []oranv1alpha1.Extension{}
+
+		extString := ExtensionToString(extensions)
+		Expect(extString).To(Equal(""))
+	})
+})
+
+var _ = Describe("DoesK8SResourceExist", func() {
+
```

Comment:
Do these tests belong here? The aren't related to extensions. May this be a copy & paste error?

Reply:
I included some extra testcases in this PR. I can add them separately in another PR.

Reply:
That is fine, keep them.